pkg/agent/agent.go

```diff
@@ -641,3 +642,26 @@ func getRoundInfo(bridgeClient ovsconfig.OVSBridgeClient) types.RoundInfo {
 
 	return roundInfo
 }
+
+func (i *Initializer) getNodeMTU(localIntf *net.Interface) int {
+	if i.networkConfig.DefaultMTU != 0 {
+		return i.networkConfig.DefaultMTU
+	}
+	mtu := localIntf.MTU
+	if mtu <= 0 {
+		panic("Failed to fetch node mtu")
```

Comment:
It would be better to return error here ( but not use "panic" directly), and let the caller to decide the error.

Reply:
I feel panic will mark the deployment in error state and user can take corresponding action. We dont want to continue if MTU cannot be discovered. This is same as PODCIDR being empty and we throw panic there

Reply:
agent cannot continue if almost any step in `Initializer` fails, for example the steps before `getNodeMTU`: `GetNodeName`, `GetNodeAddr`, `GetIPNetDeviceFromIP`. I agree with @wenyingd we should pop up the error and handle it in same style, otherwise there could be quite a lot panic in the program.
Besides, I think the errors won't get into log files by `panic`.

Reply:
Done